### PR TITLE
subcategories

### PR DESCRIPTION
**Pull request by tehnalogos**

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/topic-list.hbs [hunk 1]
```diff
@@ -1,167 +1,169 @@
...
-            so it may vary per env and needs to be validated in the user_custom_fields table }}
-            {{#if this.currentUser.custom_fields.user_field_1}}
-              <div>Organisation:
-                <strong>{{this.currentUser.custom_fields.user_field_1}}</strong>
-              </div>
-            {{/if}}
-            {{#if this.currentUser.custom_fields.user_field_2}}
-              <div>Region:
-                <strong>{{this.currentUser.custom_fields.user_field_2}}</strong>
+{{#if this.showTopicsList}}
```
Comment (tehnalogos) [inline app/assets/javascripts/discourse/app/components/topic-list.hbs:1]: this line is the only thing that should have changed in this file

File: app/assets/javascripts/discourse/app/components/d-navigation.js [hunk 2]
```diff
@@ -11,6 +11,20 @@ export default Component.extend(FilterModeMixin, {
   router: service(),
   dialog: service(),
   tagName: "",
+  canShowTopicButton: false,
+
+  didInsertElement() {
```
Comment (tehnalogos) [inline app/assets/javascripts/discourse/app/components/d-navigation.js:16]: only show create topic button on bottom most category pages

File: app/assets/javascripts/discourse/app/components/topic-list.js [hunk 3]
```diff
@@ -52,6 +53,12 @@ export default Component.extend(LoadMore, {
       });
   },
 
+  canShowTopicsList() {
+    const canCreateTopic =
+      this.canCreateTopic === undefined || this.canCreateTopic;
```
Comment (tehnalogos) [inline app/assets/javascripts/discourse/app/components/topic-list.js:58]: sometimes canCreateTopic comes back as undefined

File: app/assets/javascripts/discourse/app/components/topic-list.js [hunk 4]
```diff
@@ -28,6 +28,7 @@ export default Component.extend(LoadMore, {
   saveButtonText: "Save",
   resetButtonText: "Reset",
   isSaving: false,
+  showTopicsList: false,
```
Comment (tehnalogos) [inline app/assets/javascripts/discourse/app/components/topic-list.js:31]: show topics list only on bottom most category pages

File: app/assets/javascripts/discourse/app/controllers/discovery/topics.js [hunk 5]
```diff
@@ -170,6 +170,9 @@ const controllerOpts = {
 
     const category = this.category;
     if (category) {
+      if (category.has_children) {
```
Comment (tehnalogos) [inline app/assets/javascripts/discourse/app/controllers/discovery/topics.js:173]: dont return prompt to create topics if category is not at bottom most level

File: app/assets/javascripts/discourse/app/templates/discovery/topics.hbs [hunk 6]
```diff
@@ -116,33 +116,5 @@
       @showDismissRead={{this.showDismissRead}}
       @resetNew={{action "resetNew"}}
     />
-
-    <FooterMessage
```
Comment (tehnalogos) [inline app/assets/javascripts/discourse/app/templates/discovery/topics.hbs:120]: get rid of footer prompt to create topic

File: app/assets/javascripts/discourse/app/components/d-navigation.js [hunk 7]
```diff
@@ -11,6 +11,20 @@ export default Component.extend(FilterModeMixin, {
...
+    this.set("canShowTopicButton", this.canCreateTopic);
```
Comment (Ledespana) [inline app/assets/javascripts/discourse/app/components/d-navigation.js]: why is this relaying on this.canCreateTopic rather than canShowTopicAction()?

Comment (tehnalogos) [reply app/assets/javascripts/discourse/app/components/d-navigation.js]: @Ledespana this.canCreateTopic is how that used to be determined by default

Comment (Ledespana) [reply app/assets/javascripts/discourse/app/components/d-navigation.js]: do you need the didInsertElement? does this.canCreateTopic rely on something in the DOM?
Can't you in line 14 do the following and avoid using didInsertElement?
```
...
canShowTopicButton:  this.canCreateTopic,
```

Comment (tehnalogos) [reply app/assets/javascripts/discourse/app/components/d-navigation.js]: @Ledespana will try this

Comment (tehnalogos) [reply app/assets/javascripts/discourse/app/components/d-navigation.js]: @Ledespana addressed using an earlier suggestion

**Review by Ledespana — APPROVED** [review]

**Pull request merged**